numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -497,21 +498,33 @@ PyArray_SelectkindConverter(PyObject *obj, NPY_SELECTKIND *selectkind)
 static int searchside_parser(char const *str, Py_ssize_t length, void *data)
 {
     NPY_SEARCHSIDE *side = (NPY_SEARCHSIDE *)data;
+    int success = -1;
 
     if (length < 1) {
         return -1;
     }
     else if (str[0] == 'l' || str[0] == 'L') {
         *side = NPY_SEARCHLEFT;
-        return 0;
+        success = 0;
```

Comment:
I think this would be clearer as:
```
is_exact = (length == 4 && strcmp(str, "left") == 0);
```
then you can avoid doing multiple `strcmp`s when you only really need one.

You can keep the `-1` return in the error case, and then the if below becomes simply `if (!is_exact)`